scripts/update-registry.ts

```diff
@@ -0,0 +1,272 @@
+import { promises as fs } from 'node:fs';
+import path from 'node:path';
+import util from 'node:util';
+
+import type { AddressType, SingletonDeploymentJSON } from '../src/types';
+
+type Options = {
+  version: string;
+  chainId: string;
+  deploymentType: string;
+  verbose: boolean;
+};
+
+/**
+ * Parses version string to extract version and deployment type
+ * Examples:
+ * - "v1.3.0-canonical" -> { version: "v1.3.0", deploymentType: "canonical" }
+ * - "v1.3.0-eip155" -> { version: "v1.3.0", deploymentType: "eip155" }
+ * - "v1.4.1" -> { version: "v1.4.1", deploymentType: "canonical" }
+ * - "v1.5.0" -> { version: "v1.5.0", deploymentType: "canonical" }
+ */
+function parseVersion(versionString: string): { version: string; deploymentType: string } {
+  const parts = versionString.split('-');
+  if (parts.length === 2) {
+    return {
+      version: parts[0],
+      deploymentType: parts[1],
+    };
+  }
+  return {
+    version: versionString,
+    deploymentType: 'canonical',
+  };
+}
+
+function parseOptions(): Options {
+  const options = {
+    version: { type: 'string' },
+    chainId: { type: 'string' },
+    verbose: { type: 'boolean' },
+  } as const;
+  const { values } = util.parseArgs({ options });
+
+  if (values.version === undefined) {
+    throw new Error('missing --version flag');
+  }
+  if (values.chainId === undefined) {
+    throw new Error('missing --chainId flag');
+  }
+
+  const { version, deploymentType } = parseVersion(values.version as string);
+
+  return {
+    version,
+    chainId: values.chainId as string,
+    deploymentType,
+    verbose: values.verbose === true,
+  };
+}
+
+/**
+ * Sorts deployment types array according to official policy:
+ * 1. If there already is a deployment for the chain, keep it first (to avoid breaking changes)
+ * 2. If deploying both variants, prefer eip155
+ */
+function sortDeploymentTypes(types: AddressType[], existingValue?: AddressType): AddressType[] {
+  // Policy #1: If there's an existing deployment, preserve it as first
+  if (existingValue !== undefined && types.includes(existingValue)) {
```

Comment:
This function is only ever called with this path - I don't think you need it at all.